### ISSUE TITLE
[QUESTION] Best way to port some of this to the web.

### ISSUE THREAD

**Issue by QuantumInformation** [bartonhammond/snowflake#28]
Hi this has inspired me to have something similar for the web. I was wondering what would be the most effective way to port some of the to a version using http://www.getmdl.io/started/ and how much of it could be ported?

**Comment by jhabdas** [comment]
@QuantumInformation Consider looking at the JS community. Right now Webpack is the premier module bundler for the Web. If you'd like to see how it can be integrated with a React Native app check out a few other [React Native Boilerplates](http://habd.as/awesome-react-boilerplates/) using Webpack. Not included in the Native list due to it's [Universal scope](https://medium.com/@mjackson/universal-javascript-4761051b7ae9) is [Este.js](https://github.com/este/este), which is one of a plethora of options for taking your app cross-platform and to the Web.

**Comment by bartonhammond** [comment]
Um..I thought I responded to this the other day...I don't know where my response went!  Sorry if you thought I was ignoring you @QuantumInformation 

So yes, I've had it on my radar to have a universal app, similiar to the [https://github.com/este/este](https://github.com/este/este) project

My concern is that this project becomes a monster.  Right now it's somewhat contained as Parse.com can be replace fairly easy.

I'm toying with a companion project to this that is made up of the following

* **React & React Native** - Good enough for Facebook, gefm (good enough for me)
* **Mongodb** - Seems a good fit
* **Redux** - Best state/action management solution
* **Hapi** - Configuration over coding - Good enough for Walmart, gefm
* Openshift - Hosting for free - Develop local, deply to the cloud - I've already worked through getting a NodeJS/Mongodb app running both locally and on OpenShift.  

Though Parse.com is good for data access it's horrible for Cloud Code as they have their own HTTP request and Promise implementation.  Some server code I have now works locally with Node but I would have to rewrite it to make it work w/ Parse.com

My next project will have a running node server (Hapi) with Register, Login, Reset Password, Logout and Protected Page - just like Snowflake but everything contained in the code (rather then in Parse.com) And it will be agnostic to the UI.  

I've found some example simple start projects that I am studying now.  
* [https://github.com/rkovacevic/hapijs-starter](https://github.com/rkovacevic/hapijs-starter)
  * React, Redux and Hapi but uses Sequelize 
  * Also, Babel ES6  Webpack - I need to understand this whole WebPack thing still

* [https://github.com/stoianmarius/hapi-token](https://github.com/stoianmarius/hapi-token)
  * Working Json Web Token implementation
  * Works w/ Mongodb

@QuantumInformation as to [http://www.getmdl.io/](http://www.getmdl.io/) I'd rather the solution be as bare of CSS as possible - everyone has their favorite CSS weapon of choice and I just want to provide something easily modified - the less opinionated the better.  ;-

@QuantumInformation @jhabdas  What do you guys think?  What are your opinions of 
1) upgrading this project to said stack 
vs
2) start new project and leave this to it's own life?

My next goal, which I need myself anyway, is to have a fully functional starter Nodejs app with User Authentication that can be run and tested locally and pushed to cloud without modification.

**Comment by jhabdas** [comment]
2

**Comment by jhabdas** [comment]
Here's a Hapi integration with Webpack. IMHO starter kits should go with Express to be more vanilla:

https://github.com/luandro/hapi-universal-redux

**Comment by bartonhammond** [comment]
@jhabdas Thanks, looks like an interesting project.

**Comment by QuantumInformation** [comment]
I think its a good idea to leave this project as a specialised parse react-native app. 

Is cloud code really that bad? I don't have much experience with it. I'm not sure its worth ditching as the overhead of maintaining your own backend it to much of a burden for me atm.

**Comment by QuantumInformation** [comment]
@bartonhammond Although I would like to see how you do the login screen with vanilla css.

**Comment by bartonhammond** [comment]
@QuantumInformation The problem when working with Parse.com Cloud Code begins when you realize there is no debugging environment available.  So you're forced to put ```console.log``` statements everywhere.   I did all that on a different project and it was terrible - painstaking slow trying to figure something out.  

As I thought about this more yesterday, I realized that snowflake was created 'cause I needed it.  This next project is taking snowflake to the next step, of having a local server & db and being able to deploy it to the cloud.  I need that now to move forward w/ my personal project.  

And I don't need a web interface - the only things I need to know about are the logs and database records.  With OpenShift I have access to the logs through ssh and database with RockMongo.  So I don't really need a React web interface for my next step.

If someone wants to add in a React UI, that would be fine but I don't have the need nor bandwidth to add that in myself.
